pkg/cloud/services/compute/instance.go

```diff
@@ -46,58 +48,149 @@ const (
 	timeoutInstanceDelete       = 5 * time.Minute
 )
 
-// constructNetworks builds an array of networks from the network, subnet and ports items in the instance spec.
-// If no networks or ports are in the spec, returns a single network item for a network connection to the default cluster network.
-func (s *Service) constructNetworks(openStackCluster *infrav1.OpenStackCluster, instanceSpec *InstanceSpec) ([]infrav1.Network, error) {
-	trunkRequired := false
+// normalizePortTarget ensures that the port has a network ID.
+func (s *Service) normalizePortTarget(port *infrav1.PortOpts, openStackCluster *infrav1.OpenStackCluster, portIdx int) error {
```

Comment:
`normalizePortTarget()` sounds a bit confusing if it's just `normalizePortNetwork()`.

Reply:
I would update this, but I plan to refactor this code a fair bit anyway after backporting this to 0.7.